rules/recording.go

```diff
@@ -113,6 +114,16 @@ func (rule RecordingRule) String() string {
 	return string(byt)
 }
 
+// UpdateEvalTimeSeconds updates evaluationTimeSeconds to the time in seconds it took to evaluate the rule on its last evaluation.
+func (rule *RecordingRule) UpdateEvalTimeSeconds(seconds float64) {
+	rule.evaluationTimeSeconds = seconds
+}
+
+// EvalTimeSeconds returns the time in seconds it took to evaluate the recording rule.
+func (rule *RecordingRule) EvalTimeSeconds() float64 {
```

Comment:
Is there a race here that warrants a lock?

Reply:
Added locks for reading and updating both the recording and alert rule's evaluation times.

Reply:
@Gouthamve Is there any further changes to be made here?